bundle/phases/deploy.go

```diff
@@ -227,5 +231,92 @@ func Deploy(ctx context.Context, b *bundle.Bundle, outputHandler sync.OutputHand
+		return v, nil
+	})
+	if err != nil {
+		log.Debugf(ctx, "failed to count resources: %s", err)
+	}
+
+	jobsIds := make([]string, 0)
+	for _, job := range b.Config.Resources.Jobs {
+		// Do not include missing IDs in telemetry. We can still detect them
+		// by comparing against the resource count.
+		if job == nil || job.ID == "" {
+			continue
+		}
+		jobsIds = append(jobsIds, job.ID)
+	}
+	pipelineIds := make([]string, 0)
+	for _, pipeline := range b.Config.Resources.Pipelines {
+		// Do not include missing IDs in telemetry. We can still detect them
+		// by comparing against the resource count.
+		if pipeline == nil || pipeline.ID == "" {
+			continue
+		}
+		pipelineIds = append(pipelineIds, pipeline.ID)
+	}
+	clusterIds := make([]string, 0)
+	for _, cluster := range b.Config.Resources.Clusters {
+		// Do not include missing IDs in telemetry. We can still detect them
+		// by comparing against the resource count.
+		if cluster == nil || cluster.ID == "" {
+			continue
+		}
+		clusterIds = append(clusterIds, cluster.ID)
+	}
+	dashboardIds := make([]string, 0)
+	for _, dashboard := range b.Config.Resources.Dashboards {
+		// Do not include missing IDs in telemetry. We can still detect them
+		// by comparing against the resource count.
+		if dashboard == nil || dashboard.ID == "" {
+			continue
+		}
+		dashboardIds = append(dashboardIds, dashboard.ID)
+	}
+
+	// sort the IDs to make the record generated deterministic
+	// this is important for testing purposes
+	slices.Sort(jobsIds)
+	slices.Sort(pipelineIds)
+	slices.Sort(clusterIds)
+	slices.Sort(dashboardIds)
+
+	// If the bundle UUID is not set, we use a default 0 value.
+	bundleUuid := "00000000-0000-0000-0000-000000000000"
+	if b.Config.Bundle.Uuid != "" {
+		bundleUuid = b.Config.Bundle.Uuid
+	}
+
+	telemetry.Log(ctx, protos.DatabricksCliLog{
+		BundleDeployEvent: &protos.BundleDeployEvent{
+			BundleUuid: bundleUuid,
+
+			ResourceCount:                     resourcesCount,
+			ResourceJobCount:                  int64(len(b.Config.Resources.Jobs)),
+			ResourcePipelineCount:             int64(len(b.Config.Resources.Pipelines)),
+			ResourceModelCount:                int64(len(b.Config.Resources.Models)),
+			ResourceExperimentCount:           int64(len(b.Config.Resources.Experiments)),
+			ResourceModelServingEndpointCount: int64(len(b.Config.Resources.ModelServingEndpoints)),
+			ResourceRegisteredModelCount:      int64(len(b.Config.Resources.RegisteredModels)),
+			ResourceQualityMonitorCount:       int64(len(b.Config.Resources.QualityMonitors)),
+			ResourceSchemaCount:               int64(len(b.Config.Resources.Schemas)),
+			ResourceVolumeCount:               int64(len(b.Config.Resources.Volumes)),
+			ResourceClusterCount:              int64(len(b.Config.Resources.Clusters)),
+			ResourceDashboardCount:            int64(len(b.Config.Resources.Dashboards)),
+			ResourceAppCount:                  int64(len(b.Config.Resources.Apps)),
+
+			ResourceJobIDs:       jobsIds,
```

Comment:
Why do we send jobIds and other ids?

Do we want to place a limit on the number of ids we send here, in case there too many?

Reply:
> Why do we send jobIds and other ids?

This gives us one table with all of the IDs associated with a bundle which makes ETL easier.

> Do we want to place a limit on the number of ids we send here, in case there too many?

I do not anticipate us running into issues with too many ids in our tables.  If that does turn out to be a concern we can always scale down our tracking for future records.

Reply:
the issue could the request timing out due to being too large rather than deliverying partial results. Putting a cap is a good practice.

Reply:
Good point. Added a cap of a 1000 ids per resource IDs.

We can scale that up after gathering some initial data and observing that this indeed is a limitation.